src/DynamoCoreWpf/HelixRenderPackage.cs

```diff
@@ -0,0 +1,577 @@
+        {
+            var points = new PointGeometry3D()
+            {
+                Positions = new Vector3Collection(),
+                Indices = new IntCollection(),
+                Colors = new Color4Collection()
+            };
+
+            return points;
+        }
+
+        internal static MeshGeometry3D InitMeshGeometry()
+        {
+            var mesh = new MeshGeometry3D()
+            {
+                Positions = new Vector3Collection(),
+                Indices = new IntCollection(),
+                Colors = new Color4Collection(),
+                Normals = new Vector3Collection(),
+                TextureCoordinates = new Vector2Collection()
+            };
+
+            return mesh;
+        }
+
+        internal static BillboardText3D InitText3D()
+        {
+            var text3D = new BillboardText3D();
+
+            return text3D;
+        }
+
+        internal static string CleanTag(string tag)
+        {
+            var splits = tag.Split(':');
+            return splits.Count() <= 1 ? "[0]" : string.Format("[{0}]",string.Join(",", splits.Skip(1)));
+        }
+
+        private static Color4 Color4FromBytes(byte red, byte green, byte blue, byte alpha)
+        {
+            var color = new Color4((red / 255.0f),(green / 255.0f),(blue / 255.0f), (alpha/255.0f));
+            return color;
+        }
+
+        private static Vector3 Vector3ForYUp(double x, double y, double z)
+        {
+            return new Vector3((float)x, (float)z, (float)-y);
+        } 
+    }
+
+    internal static class HelixRenderExtensions
+    {
+        public static double[] ToDoubleArray(this Vector2Collection collection)
+        {
+            var doubles = new double[collection.Count*2];
+
+            var count = 0;
+            foreach (var v in collection)
+            {
+                doubles[count] = v.X;
+                doubles[count + 1] = v.Y;
+                count += 2;
+            }
+
+            return doubles;
+        }
+
+        public static double[] ToDoubleArray(this Vector3Collection collection)
+        {
+            var doubles = new double[collection.Count*3];
+
+            var count = 0;
+            foreach (var v in collection)
+            {
+                doubles[count] = v.X;
+                doubles[count + 1] = v.Y;
+                doubles[count + 2] = v.Z;
+                count += 3;
+            }
+
+            return doubles;
+        }
+
+        public static byte[] ToByteArray(this Color4Collection collection)
+        {
+            var bytes = new byte[collection.Count * 4];
+
+            var count = 0;
+            foreach (var c in collection)
+            {
+                bytes[count] = (byte)(c.Red * 255.0f);
+                bytes[count + 1] = (byte)(c.Green * 255.0f);
+                bytes[count + 2] = (byte)(c.Blue * 255.0f);
+                bytes[count + 3] = (byte)(c.Alpha * 255.0f);
+                count += 4;
+            }
+
```

Comment:
This doesn't need to build an array.  Just use the yield statement.  This means you're not copying a huge amount of data.